Package splitting and documentation updates (#19)
juliendelplanque/JRPC: docs/MessageProcessing.md

# Message Processing

JSON RCP messages are processed by instances of `JRPCMessageProcessor` independently of the underlying transport used for receiving and responding the messages.

> The message processor has available a list of method handlers that will be searched when a incoming message is received, if a suitable handler is found then it gets executed with the parameters provided in the message.

Reviewer: ```suggestion
The message processor has a list of method handlers available that will be searched when a incoming message is received. If a suitable handler is found (the `#methodName` is matching) then it gets executed with the parameters provided in the message.
```

> To configure a message processor use any of the following messages to add new handlers:

Reviewer: ```suggestion
A message processor can be configured using the following messages:
```

> - `addHandlerNamed:block:` will add a new handler for a method that will evaluate the provided block

Reviewer: ```suggestion
- `addHandlerNamed:block:`adds a new handler for a method that will evaluate the block provided as argument.
```

> - `addHandlersFromPragmasIn:` will search for `<jrpc>` pragmas declared in the methods of the parameter object, and will add a message handler if found.

Reviewer: ```suggestion
- `addHandlersFromPragmasIn:` searches for `<jrpc>` pragmas declared in the methods of the object provided as argument, and adds a message handler if such pragma is found. When such handler is added, the method in which the pragma was found is sent to the object when a matching JRPC message is received.
```

> - `addHandler:` will add a new handler. Handlers must subclass `JRPCAbstractHandler`, this allows the end user to extend the available options if needed.

Reviewer: ```suggestion
- `addHandler:` adds a new handler. Handlers must subclass `JRPCAbstractHandler`, this allows the end user to extend the available options if needed.
```

> The message processor main API is `handleJSON:` , any server implementing an specific transport protocol must obtain the json string and delegate it's processing by sending this message and will receive the json string to be used as response.

Reviewer: ```suggestion
The message processor main API is `handleJSON:`, any server implementing a specific transport protocol must obtain the json string and delegate its processing by sending `handleJSON:` to the processor. `handleJSON:` method returns the JSON string to return to the client once the processing is completed.
```

## Message handlers

> Message handlers are responsible of executing the code related to a method. All handlers are subclasses of `JRPCAbstractHandler` and need to provide a `methodName` and a way to execute this method with the provided arguments. There's two implementations already available `JRPCBlockHandler` and `JRPCPragmaHandler`.

Reviewer: ```suggestion
Message handlers are responsible of executing the code related to a JRPC method. All handlers are subclasses of `JRPCAbstractHandler` and need to provide a `methodName` and a way to execute this method with the provided arguments. There's two implementations already available `JRPCBlockHandler` and `JRPCPragmaHandler`.
```

> - `JRPCBlockHandler` instances are created with a block closure that will we evaluated to compute the method. The parameter names are taken from the block.

Reviewer: ```suggestion
- `JRPCBlockHandler` instances are created with a block closure that will be evaluated to compute the method. The parameter names are taken from the block parameters names.
```
